### PR TITLE
feat: add aliasfix to fix migrated import paths

### PR DESCRIPTION
**Pull request by codyoss**
This will assist us in the migration effort and is also a tool our users can use.
The tool is allowlisted to only migrate specific imports, and only if we have
made a configuration change to say the code has been migrated.

### CODE REVIEW

File: internal/aliasfix/main.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"golang.org/x/tools/imports"
+)
+
+var (
+	fset = token.NewFileSet()
+)
+
+func main() {
+	flag.Parse()
+	path := flag.Arg(0)
```
Comment (noahdietz) [inline internal/aliasfix/main.go:44]: Please provide an error message if the argument isn't provided :)

File: internal/aliasfix/main.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		})
+	} else {
+		if err := processFile(path, nil); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func processFile(name string, w io.Writer) error {
```
Comment (noahdietz) [inline internal/aliasfix/main.go:73]: Nit: Can you add a comment that `w io.Writer` is present in the signature for test accessibility ? I was wondering to myself why it was there until I got to the test 😅

**Review by noahdietz — COMMENTED** [review]
FWIW, the GAPIC generator already does this in that if we give it an "import path override" for the protos its generating, it will know to generate request/response/stub types using the "overridden" import path. This of course does not help with manual clients.

I do realize though that there has to be a time where we generate the stubs in google-cloud-go _without_ changing the type in the signature for the GAPIC, while we do the alias/release dance. Perhaps we need to toggle when the GAPIC generator is made aware of he import path override e.g. decouple it from configuring gapicgen to generate stubs in google-cloud-go. Right now it is coupled (see storage) in that if we want to provide an override, the GAPIC will respect it rather than deferring to go-genproto still.

**Comment by codyoss** [comment]
@noahdietz This is supposed to be more of an equivalent to `go fix`. I fully intend for us to leverage the generator override. This is for updating existing code that references go-genproto.

For example: 
- Users can run this command in their local code to fixup all import paths automatically.
- We can use this in the samples repo to mass update samples to use new type locations.

File: internal/aliasfix/main.go (hunk 1 above)
Comment (codyoss) [reply internal/aliasfix/main.go:44]: done.

File: internal/aliasfix/main.go (hunk 2 above)
Comment (codyoss) [reply internal/aliasfix/main.go:78]: done.

**Comment by codyoss** [comment]
> I do realize though that there has to be a time where we generate the stubs in google-cloud-go without changing the type in the signature for the G[API](https://docs.github.com/)C, while we do the alias/release dance. Perhaps we need to toggle when the GAPIC generator is made aware of he import path override e.g. decouple it from configuring gapicgen to generate stubs in google-cloud-go. Right now it is coupled (see storage) in that if we want to provide an override, the GAPIC will respect it rather than deferring to go-genproto still.

You are right. We may be able to accomplish them by just having a toggle in gapicgen to regen in both locations in two different invocations of protoc

**Comment by noahdietz** [comment]
> @noahdietz This is supposed to be more of an equivalent to `go fix`. I fully intend for us to leverage the generator override. This is for updating existing code that references go-genproto.
> 
> For example:
> 
> * Users can run this command in their local code to fixup all import paths automatically.
> * We can use this in the samples repo to mass update samples to use new type locations.

Oh! cool! awesome.

**Review by noahdietz — APPROVED** [review]

**Pull request merged**